### PR TITLE
Cart mgmt

### PR DESCRIPTION
**Pull request by Meghana-sk**
Cart management flow

1. Add item to cart from products page after login
    To login please use `email: adarshbalika@gmail.com` and `password: adarshbalika`
3. After adding an item, Add to cart button changes to Go to cart, upon click navigates directly to Cart page
4. Display number of items in cart on navbar
5. Fetch cart items from db
6. Increase/Decrease quantity of product from Cart page
7. Remove items from cart
8.  Cart summary and price details regarding

1. Move to wishlist on product card as well as card in Cart page

### CODE REVIEW

**Review by mohammed-sas — APPROVED** [review]
Ading items to cart, removing items from cart and changing the quantity of product is working as intended
Cart context has been written well
Cose is clean and readable

**Review by kart-c — CHANGES_REQUESTED** [review]
Can you fix this? Cart item still remains even if the quantity is 0

**Comment by AnkurChunekar** [comment]
- When Signing Up when user enters exact 8 characters even when confirm and password  is same, user is alerted with a warning that both should be same, but as soon as I enter more that 8 chars I was able to sign in, this creates a lot of confusion, please fix this.
- Change your Validation alert mesaage to `Password should be more than 8 characters`  instead of `Password should be min 8 characters`

File: src/context/cart-context.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        { product: item },
+        {
+          headers: { authorization: authState.token },
+        }
+      );
+      cartDispatch({ type: ADD_TO_CART, payload: response.data.cart });
+    } catch (err) {
+      alert(err);
+    }
+  };
```
Comment (Jay-Gadhiya) [inline src/context/cart-context.js:47]: you can extract all this function in new file

**Comment by AnkurChunekar** [comment]
If I add relatively higher number products in the cart, I have to scroll alot just to check the price details card.
Maybe because you have aligned the items in center for cart container, aligning them to start would fix this.
```
.prod-select-container {
align-items: start;
}
```

**Review by Jay-Gadhiya — APPROVED** [review]
- code looks clean
- you can fix ui issue suggested by kart-c

**Comment by AnkurChunekar** [comment]
meghana pr

- After removing all the products, even when i have 0 products, Price details card displays 2 items in cart. please look into this.
- After fixing the above bug, you can only display the price details card when cart contains at least 1 product.

**Review by kart-c — APPROVED** [review]

File: src/context/cart-context.js (hunk 1 above)
Comment (Meghana-sk) [reply src/context/cart-context.js:47]: > you can extract all this function in new file

Will do in code refactoring PR

**Review by AnkurChunekar — APPROVED** [review]
- All the bugs have been fixed
- possible edge cases have handled.
- Code looks good to merge
- great work @Meghana-sk

**Comment by Meghana-sk** [comment]
> Can you fix this? Cart item still remains even if the quantity is 0 

Fixed

**Comment by Meghana-sk** [comment]
> meghana pr
> 
> * After removing all the products, even when i have 0 products, Price details card displays 2 items in cart. please look into this.
> * After fixing the above bug, you can only display the price details card when cart contains at least 1 product.

fixed

**Pull request merged**